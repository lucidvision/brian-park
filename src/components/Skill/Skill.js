import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { HeadingTertiary, Paragraph } from 'components'
import { utilities } from 'styles'

Skill.propTypes = {
  content: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

Skill.defaultProps = {
  content: '',
  title: ''
}

const StyledDiv = styled.div`
  padding: 2.5rem;
  display: flex;
  flex-direction: column;
  border-radius: 3px;
  background-color: rgba(255, 255, 255, .8);
  box-shadow: 0 1.5rem 4rem rgba(0, 0, 0, .15);

  &:hover {
    transform: scale(1.03);
  }

  & > h3 {
    margin-bottom: ${utilities.marginSmall};
  }
`

export default function Skill (props) {
  const { content, title, ...rest } = props
  return (
    <StyledDiv {...rest}>
      <HeadingTertiary text={title} />
      <Paragraph text={content} />
    </StyledDiv>
  )
}
