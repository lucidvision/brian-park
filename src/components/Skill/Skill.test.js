import React from 'react'
import { mount } from 'enzyme'
import Skill from './Skill'

const createSkill = (override) => {
  const props = {
    content: 'Rainbows',
    title: 'Unicorns',
    ...override}

  return mount(<Skill {...props} />)
}

describe('Skill', () => {
  const component = createSkill()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(component.find('Skill').props()).toEqual({
      content: 'Rainbows',
      title: 'Unicorns'
    })
  })

  it('renders a tertiary header with the correct text', () => {
    expect(component.find('HeadingTertiary').props()).toMatchObject({
      text: 'Unicorns'
    })
  })

  it('renders a paragraph with the correct text', () => {
    expect(component.find('Paragraph').props()).toMatchObject({
      text: 'Rainbows'
    })
  })
})
