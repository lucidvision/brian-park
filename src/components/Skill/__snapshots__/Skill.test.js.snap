// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill renders correctly 1`] = `
<Skill
  content="Rainbows"
  title="Unicorns"
>
  <styled.div>
    <div
      className="sc-gqjmRU kldQLE"
    >
      <HeadingTertiary
        color="rgba(0, 13, 85, 0.8)"
        text="Unicorns"
      >
        <styled.h3
          color="rgba(0, 13, 85, 0.8)"
        >
          <h3
            className="sc-bZQynM jZhXzy"
            color="rgba(0, 13, 85, 0.8)"
          >
            Unicorns
          </h3>
        </styled.h3>
      </HeadingTertiary>
      <Paragraph
        color="rgba(0, 13, 85, 0.8)"
        size="1.8rem"
        text="Rainbows"
      >
        <styled.p
          color="rgba(0, 13, 85, 0.8)"
          size="1.8rem"
        >
          <p
            className="sc-iwsKbI iOZGO"
            color="rgba(0, 13, 85, 0.8)"
            size="1.8rem"
          >
            Rainbows
          </p>
        </styled.p>
      </Paragraph>
    </div>
  </styled.div>
</Skill>
`;
